### PR TITLE
feat: add goerli flashbots scripts

### PR DESCRIPTION
**Pull request by lbertenasco**

### CODE REVIEW

File: scripts/flashbots/02-goerli-send-tx.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    await mainExecute();
+  });
+}
+
+function mainExecute(): Promise<void | Error> {
+  return new Promise(async (resolve, reject) => {
+    const [owner] = await ethers.getSigners();
+    console.log('using address:', owner.address);
+
+    const provider = ethers.providers.getDefaultProvider('goerli');
```
Comment (b0dhidharma) [inline scripts/flashbots/02-goerli-send-tx.ts]: Be careful, this actually doesnt use the provider URL you give to it. You should use `ethers.provider` afaik, and execute the script with `--network goerli`.

File: scripts/flashbots/02-goerli-send-tx.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    const stealthRelayer = await ethers.getContractAt('StealthRelayer', contracts.stealthRelayer.goerli);
+    const stealthERC20 = await ethers.getContractAt('StealthERC20', contracts.stealthERC20.goerli);
+    console.log('creating signer');
+    const signer = new ethers.Wallet(('0x' + process.env.GOERLI_PRIVATE_KEY) as string).connect(provider) as Signer;
+
+    // `authSigner` is an Ethereum private key that does NOT store funds and is NOT your bot's primary key.
+    // This is an identifying key for signing payloads to establish reputation and whitelisting
+    // In production, this should be used across multiple bundles to build relationship. In this example, we generate a new wallet each time
+    console.log('creating flashbotSigner');
+    const flashbotSigner = new ethers.Wallet(('0x' + process.env.FLASHBOTS_PRIVATE_KEY) as string).connect(provider) as Signer;
```
Comment (b0dhidharma) [inline scripts/flashbots/02-goerli-send-tx.ts]: the shievers. this ethers feature needs to be done asap.

File: scripts/flashbots/02-goerli-send-tx.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    // build bundle
+    const bundle = [
+      {
+        signedTransaction,
+      },
+    ];
+    const signedBundle = await flashbotsProvider.signBundle(bundle);
+    let simulation: SimulationResponse;
+    try {
+      simulation = await flashbotsProvider.simulate(signedBundle, blockNumber + 1);
```
Comment (b0dhidharma) [inline scripts/flashbots/02-goerli-send-tx.ts:95]: Interesting ...

File: scripts/flashbots/02-goerli-send-tx.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      } else {
+        console.log(error);
+      }
+      return;
+    }
+    if ('error' in simulation) {
+      console.log(`Simulation Error: ${simulation.error.message}`);
+    } else {
+      console.log(`Simulation Success: ${JSON.stringify(simulation, null, 2)}`);
+    }
```
Comment (b0dhidharma) [inline scripts/flashbots/02-goerli-send-tx.ts]: Should this have a return ?

**Review by b0dhidharma — APPROVED** [review]
lg2m ! Nice work ! It's awesome that we are adding this knowledge to yMechs. Additionally: you rolled back into `npm`, should be using `yarn` (pipeline already fixed)

File: scripts/flashbots/02-goerli-send-tx.ts (hunk 1 above)
Comment (lbertenasco) [reply scripts/flashbots/02-goerli-send-tx.ts]: yeah, nice catch, I'll add a check that ensures networkId is goerli's

File: scripts/flashbots/02-goerli-send-tx.ts (hunk 3 above)
Comment (lbertenasco) [reply scripts/flashbots/02-goerli-send-tx.ts:95]: simulation is cool af

**Pull request merged**